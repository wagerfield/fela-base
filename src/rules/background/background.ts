import * as CSS from "csstype"
import { Length } from "../../types"
import { getRuleProps } from "../defaults"

export interface BackgroundRuleProps {
  attachment?: CSS.BackgroundAttachmentProperty
  positionX?: CSS.BackgroundPositionXProperty<Length>
  positionY?: CSS.BackgroundPositionXProperty<Length>
  position?: CSS.BackgroundPositionProperty<Length>
  origin?: CSS.BackgroundOriginProperty
  repeat?: CSS.BackgroundRepeatProperty
  color?: CSS.BackgroundColorProperty
  image?: CSS.BackgroundImageProperty
  clip?: CSS.BackgroundClipProperty
  size?: CSS.BackgroundSizeProperty<Length>
}

export const NAME = "background"
export const DEFAULTS: BackgroundRuleProps = {}

export default (props?: BackgroundRuleProps) => {
  const p = getRuleProps(NAME, DEFAULTS, props)
  return {
    backgroundAttachment: p.attachment,
    backgroundPositionX: p.positionX,
    backgroundPositionY: p.positionY,
    backgroundPosition: p.position,
    backgroundOrigin: p.origin,
    backgroundRepeat: p.repeat,
    backgroundColor: p.color,
    backgroundImage: p.image,
    backgroundClip: p.clip,
    backgroundSize: p.size
  }
}
