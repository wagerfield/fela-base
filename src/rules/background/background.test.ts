import background from "./background"

describe("background(props)", () => {
  it("renders expected snapshots", () => {
    expect(background()).toMatchSnapshot()
    expect(
      background({
        color: "red"
      })
    ).toMatchSnapshot()
  })
})
