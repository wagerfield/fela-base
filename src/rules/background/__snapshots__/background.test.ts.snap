// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`background(props) renders expected snapshots 1`] = `
Object {
  "backgroundAttachment": undefined,
  "backgroundClip": undefined,
  "backgroundColor": undefined,
  "backgroundImage": undefined,
  "backgroundOrigin": undefined,
  "backgroundPosition": undefined,
  "backgroundPositionX": undefined,
  "backgroundPositionY": undefined,
  "backgroundRepeat": undefined,
  "backgroundSize": undefined,
}
`;

exports[`background(props) renders expected snapshots 2`] = `
Object {
  "backgroundAttachment": undefined,
  "backgroundClip": undefined,
  "backgroundColor": "red",
  "backgroundImage": undefined,
  "backgroundOrigin": undefined,
  "backgroundPosition": undefined,
  "backgroundPositionX": undefined,
  "backgroundPositionY": undefined,
  "backgroundRepeat": undefined,
  "backgroundSize": undefined,
}
`;
